### PR TITLE
Add support for deleting widgets and add keyboard-accessible way to expand/collapse widget form controls

### PR DESCRIPTION
**Pull request by TrevorMills**
Addressing issues #26 & #22.  For #26, I think there is a better way to update the wp.customize object, but I'm not familiar with the api.  See @todo in widget-customizer.js.

### CODE REVIEW

**Comment by westonruter** [comment]
@TrevorMills the Travis build is failing. If you could run the `bin/pre-commit` script in the plugin, you can determine what is failing before I get a chance to check it out. You'll need to have PHP_CodeSniffer, the [WordPress Coding Standards](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards), and JSHint installed. The first two are now [bundled with VVV](https://github.com/10up/varying-vagrant-vagrants/pull/172).

**Comment by westonruter** [comment]
@TrevorMills thanks for fixing the build!

**Comment by westonruter** [comment]
@TrevorMills oh man, this is working really well! You're right, it is very clean. Nice work! :+1: 

It seems there is one missing piece, which actually doesn't manifest itself in the frontend at all, but it is a matter of finalizing the deletion. So right now you are removing the widget ID from the `sidebars_widgets`, which is _perfect_. However, the widget instance is remaining in the widget option. For example, if you tried deleting a Text widget, the instance will remain in the `widget_text` DB option array, for example:

```
Array
(
    [2] => Array
        (
            [title] => Widget remaining
            [text] => Yes
            [filter] => 
        )

    [3] => Array
        (
            [title] => Widget removed
            [text] => Yes
            [filter] => 
        )

    [_multiwidget] => 1
)
```

I deleted the instance with the array index `3` while in the customizer, but you can see it remains here in the option. For how the existing widgets code deletes widgets and removes their instances, see [relevant code in `widgets.php`](https://github.com/WordPress/WordPress/blob/7d22195cbb87eb4ebf203dcfe9e94e675183934d/wp-includes/widgets.php#L203-L215).

However, I just had an idea: instead of deleting the widget entirely, what if instead the **Delete** link moved the widget to the `wp_inactive_widgets` sidebar, and so the widgets could be then resurrected after having been deleted? It seems to me that deleting widgets is far too easy since there is no way to undo deleting them. If we moved them to the `wp_inactive_widgets` however, at least while in the Customizer, then they would be able to undo what they had done. 

Thoughts?

File: widget-customizer.js [hunk 1]
```diff
@@ -359,4 +391,4 @@ var WidgetCustomizer = (function ($) {
 	};
 
 	return self;
-}( jQuery ));
+}( jQuery ));
```
Comment (westonruter) [inline widget-customizer.js:394]: @TrevorMills please restore the EOF EOL in both files.

**Comment by westonruter** [comment]
@TrevorMills I talked with @shaunandrews in the weekly [Widgets UI Refresh meeting](https://irclogs.wordpress.org/chanlog.php?channel=wordpress-ui&day=2013-10-14&sort=asc#m128502), and proposed that we repurpose the Inactive Widgets “sidebar” into a Widgets Trash sidebar. He supports the idea, for both in the Customizer as well as the main widgets admin page. So the **Delete** button on the main widgets page will also become a **Trash** button, and widgets removed from sidebars will always go there.

As far as how to implement this, I suppose we'll need to filter the `sidebars_widgets` save action/filter and check to see if there are any widget instances which are orphaned and not assigned to any sidebar. If there are, then they should get appended to `sidebars_widgets[wp_inactive_widgets]`. Actually, it seems [`retrieve_widgets()`](https://github.com/WordPress/WordPress/blob/1516b990f1a1848c404268be4bda5e63de5bf4cc/wp-includes/widgets.php#L1175-L1272) is coded to do this check for us, and builds up an `orphaned_widgets` sidebar which could then be automatically merged into `wp_inactive_widgets`.

He's going to work on a Customizer UI for adding new widgets and restoring trashed ones, but in the mean time I suggested that we just have a dropdown that appears at the top of any customizer sidebar section which lists all widgets available for creating new instances, and all widgets which are currently in the trash. Selecting one of the inactive widgets would remove that item from the `select` list, and restore it to the sidebar.

File: widget-customizer.js (hunk 1 above)
Comment (westonruter) [inline widget-customizer.js:394]: Addressed in 0e6d3aff627f64d211a35d408c107902bc7b9d5e

**Comment by TrevorMills** [comment]
@westonruter coming back online after the long weekend.  Thanks for forwarding the link re: the conversation in the Widgets UI Refresh Meeting.  I'm a little busy this week, but will try to find time to work on the select list.  Is anyone else active or willing to be active on this repo at this point?

**Comment by shaunandrews** [comment]
I'm here. :)

Not sure how active I can/will be, but  I'm here. 

## 

Shaun

**Comment by TrevorMills** [comment]
nice to meet you @shaunandrews.

**Pull request merged**